### PR TITLE
Global annotations

### PR DESCRIPTION
**Pull request by Dragon31337**
fixed global annotations based on WIP branch by Joshua-Ashton 
fixed visual studio build
added config option to enable debug utils in addition to DXVK_PERF_EVENTS=1

### CODE REVIEW

**Comment by Joshua-Ashton** [comment]
Hi, you should make your "fix global annotations" a fixup commit and add yourself as a co author to the commit

**Comment by Joshua-Ashton** [comment]
Also, MSVC build works on the CI, why doesnt it for you?

**Comment by Dragon31337** [comment]
Without vs build fix I get this error:

```
../src/d3d9/d3d9_device.cpp(2347): error C2593: 'operator ==' is ambiguous
d:\Projects\dxvk\src\d3d9\d3d9_util.h(217): note: could be 'bool dxvk::operator ==(const RECT &,const RECT &)'
C:\Program Files (x86)\Windows Kits\10\include\10.0.20348.0\um\d3d11.h(1321): note: or       'bool operator ==(const D3D11_RECT &,const D3D11_RECT &)' [found using argument-dependent lookup]
../src/d3d9/d3d9_device.cpp(2347): note: while trying to match the argument list '(RECT, const RECT)'
```

C compiler for the host machine: cl (msvc 19.29.30139 "Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30139 for x64")
C linker for the host machine: link link 14.29.30139.0
C++ compiler for the host machine: cl (msvc 19.29.30139 "Microsoft (R) C/C++ Optimizing Compiler Version 19.29.30139 for x64")
C++ linker for the host machine: link link 14.29.30139.0

**Comment by Dragon31337** [comment]
Looks like I got a better fix as MS SDK has a define:

**Comment by Dragon31337** [comment]
@Joshua-Ashton, the "fix global annotations" commit was merged, and I have a more robust VS build fix to define D3D11_NO_HELPERS before including Windows SDK headers.

File: src/util/util_misc.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  inline void DecodeD3DCOLOR(D3DCOLOR color, float* rgba) {
+    // Encoded in D3DCOLOR as argb
+    rgba[3] = (float)((color & 0xff000000) >> 24) / 255.0f;
+    rgba[0] = (float)((color & 0x00ff0000) >> 16) / 255.0f;
+    rgba[1] = (float)((color & 0x0000ff00) >> 8)  / 255.0f;
+    rgba[2] = (float)((color & 0x000000ff))       / 255.0f;
+  }
+
+  template <typename T, size_t n>
+  size_t countof(const T(&)[n]) { return n; }
```
Comment (Joshua-Ashton) [inline src/util/util_misc.h:16]: Can toss this countof and use std::size instead.

**Review by Joshua-Ashton — CHANGES_REQUESTED** [review]
Minor thing... Other than that seems fine (although I guess most of it is my code :frog:)

Thanks for finishing it off.

@doitsujin any comments?

File: src/util/util_misc.h (hunk 1 above)
Comment (Dragon31337) [reply src/util/util_misc.h]: this countof was never used, I dropped it.

**Pull request merged**